lib/ansible/modules/files/blockinfile.py

```diff
@@ -311,14 +311,17 @@ def main():
 
     if changed and not module.check_mode:
         if module.boolean(params['backup']) and path_exists:
-            module.backup_local(path)
+            backup_file_path = module.backup_local(path)
         write_changes(module, result, path)
 
     if module.check_mode and not path_exists:
         module.exit_json(changed=changed, msg=msg)
 
     msg, changed = check_file_attrs(module, changed, msg)
-    module.exit_json(changed=changed, msg=msg)
+    if module.boolean(params['backup']) and path_exists:
+        module.exit_json(changed=changed, msg=msg, backup=backup_file_path)
```

Comment:
the 'common key name' is `backup_file`, also i would think the 'if' would rely on backup_file_path being populated not on path_exists?

Reply:
I have modified this pull request applying the suggestion report in your review, tested it and applying the underlying modification of source code merged in devel in the meantime